src/ObjectUtils/ElfFile.cpp

```diff
@@ -54,16 +54,16 @@ class ElfFileImpl : public ElfFile {
   ElfFileImpl(std::filesystem::path file_path,
               llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary);
 
-  [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbolsFromSymtab() override;
+  [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbols() override;
   [[nodiscard]] ErrorMessageOr<ModuleSymbols> LoadSymbolsFromDynsym() override;
   [[nodiscard]] ErrorMessageOr<uint64_t> GetLoadBias() const override;
-  [[nodiscard]] bool HasSymtab() const override;
+  [[nodiscard]] bool HasSymbols() const override;
   [[nodiscard]] bool HasDynsym() const override;
   [[nodiscard]] bool HasDebugInfo() const override;
   [[nodiscard]] bool HasGnuDebuglink() const override;
   [[nodiscard]] bool Is64Bit() const override;
   [[nodiscard]] std::string GetBuildId() const override;
-  [[nodiscard]] std::string GetSoname() const override;
+  [[nodiscard]] std::string GetPrettyName() const override;
```

Comment:
soname is more like ID. I do not think pretty_name fits here. Also is there windows counterpart for this?

Reply:
I was going by how this is used. We set the "name" field in ModuleInfo to either what we get as soname, or as the filename. This is what is displayed in the UI. (Not sure if it's used elsewhere, but it seems to me that once that name is in ModuleInfo, all special meaning of it has been lost anyway.)

My suggestion would be to keep GetPrettyName(), as it's intended for UI usage (unless I'm missing something). A basic implementation would be just using the filename. 

If we need soname for ELF files with the special identifier meaning, I would suggest to add that as an extra method in ElfFile.

Reply:
It is possible soname should be a property of an elf-file.

Reply:
it is also used by dynamic instrumentation to correctly find libc/libdl

Reply:
I see, it seems we don't distinguish between soname and name of the module, at least in the user space instrumentation code. The code seems to expect that ModuleInfo::name is the soname of the module.

So there is some ambiguity in ModuleInfo, as it's not clear that "name" should be soname when a soname is present. 

Is the code in the user space instrumentation correct if the name field was filled with the filename and not the soname? Or asked differently, if an ELF file doesn't have an soname, is the filename the correct choice then?

Not sure what the best option here is: Should we just call the method GetName() (which is then using the same name as ModuleInfo)? Should ModuleInfo get a soname field? 

My suggestion:
- GetName() in base class (filename, if no better choice).
- ModuleInfo gets added soname field (it already has other ELF specific fields)
- GetSoname() in ElfFile
- ElfFile implements GetName() as GetSoname() if present, o/w as filename.
- user space instrumentation code is changed to read soname(), not name(). 

WDYT?

Reply:
sgtm!

Reply:
Done.